### ISSUE TITLE
[Installation Error] macOS Monterey can't install tslearn : ERROR: Failed building wheel for llvmlite

### ISSUE THREAD

**Issue by Mishuni** [tslearn-team/tslearn#371]
**Describe the bug**
`$ python -m pip install tslearn`

but can't install tslearn

**To Reproduce**

python version 3.8.12
pip version 21.2.4

**Expected behavior**

I just want to install tslearn in my environment
but, I can't find the reason of this installation error.

#74 I did follow this issue before
but it doesn't work

**Environment (please complete the following information):**
 - OS: macOS Monterey 12.0.1
 - Chip : Apple M1
 - xcode-select version 2392
 - apple swift version 5.5.1 (target: arm64-apple-darwin21.1.0) 
 
 The error log is below

```Collecting tslearn
  Using cached tslearn-0.5.2-cp38-cp38-macosx_11_0_arm64.whl
Requirement already satisfied: numpy in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from tslearn) (1.20.3)
Collecting numba
  Using cached numba-0.54.1.tar.gz (2.2 MB)
Requirement already satisfied: joblib in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from tslearn) (1.1.0)
Requirement already satisfied: scikit-learn in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from tslearn) (1.0.1)
Requirement already satisfied: scipy in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from tslearn) (1.7.3)
Requirement already satisfied: Cython in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from tslearn) (0.29.24)
Collecting llvmlite<0.38,>=0.37.0rc1
  Using cached llvmlite-0.37.0.tar.gz (125 kB)
Requirement already satisfied: setuptools in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from numba->tslearn) (59.2.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages (from scikit-learn->tslearn) (3.0.0)
Building wheels for collected packages: numba, llvmlite
  Building wheel for numba (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/bunnyjw/miniforge3/envs/sample/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/numba_1d21a4dc073e4d66a9e52668c75af458/setup.py'"'"'; __file__='"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/numba_1d21a4dc073e4d66a9e52668c75af458/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-wheel-m1x_zcpk
       cwd: /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/numba_1d21a4dc073e4d66a9e52668c75af458/
  Complete output (902 lines):
  TBB not found
  Using OpenMP from: /Users/bunnyjw/miniforge3/envs/sample
  running bdist_wheel
  running build
  got version from file /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/numba_1d21a4dc073e4d66a9e52668c75af458/numba/_version.py {'version': '0.54.1', 'full': '39aef3deb073a67b0bf121e099ecf5b33d799f40'}
  running build_py

   .
   .
   .
  copying numba/core/runtime/_nrt_python.c -> build/lib.macosx-11.0-arm64-3.8/numba/core/runtime
  copying numba/core/runtime/_nrt_pythonmod.c -> build/lib.macosx-11.0-arm64-3.8/numba/core/runtime
  copying numba/core/runtime/nrt.c -> build/lib.macosx-11.0-arm64-3.8/numba/core/runtime
  copying numba/core/runtime/nrt.h -> build/lib.macosx-11.0-arm64-3.8/numba/core/runtime
  copying numba/core/runtime/nrt_external.h -> build/lib.macosx-11.0-arm64-3.8/numba/core/runtime
  copying numba/core/annotations/template.html -> build/lib.macosx-11.0-arm64-3.8/numba/core/annotations
  copying numba/cuda/tests/cudadrv/data/jitlink.ptx -> build/lib.macosx-11.0-arm64-3.8/numba/cuda/tests/cudadrv/data
  running build_ext
  building 'numba._dynfunc' extension
  Warning: Can't read registry to find the necessary compiler setting
  Make sure that Python modules winreg, win32api or win32con are installed.
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8
  creating build/temp.macosx-11.0-arm64-3.8/numba
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/_dynfuncmod.c
  In file included from numba/_dynfuncmod.c:1:
  numba/_dynfunc.c:143:5: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
      0,                         /* tp_print */
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  In file included from numba/_dynfuncmod.c:1:
  numba/_dynfunc.c:253:5: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
      0,                         /* tp_print */
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  In file included from numba/_dynfuncmod.c:1:
  numba/_dynfunc.c:470:5: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
      0,                         /* tp_print */
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  3 warnings generated.
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/_dynfuncmod.o -o build/lib.macosx-11.0-arm64-3.8/numba/_dynfunc.cpython-38-darwin.so
  building 'numba._dispatcher' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/core
  creating build/temp.macosx-11.0-arm64-3.8/numba/core/typeconv
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/include -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/_dispatcher.cppgcc: numba/_typeof.c
 
  gcc: numba/core/typeconv/typeconv.cpp
  gcc: numba/_hashtable.c
  numba/_dispatcher.cpp:894:37: warning: offset of on non-POD type 'Dispatcher' [-Winvalid-offsetof]
      {(char*)"_can_compile", T_BOOL, offsetof(Dispatcher, can_compile), 0, NULL },
                                      ^                    ~~~~~~~~~~~
  /Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/stddef.h:104:24: note: expanded from macro 'offsetof'
  #define offsetof(t, d) __builtin_offsetof(t, d)
                         ^                     ~
  numba/_dispatcher.cpp:953:5: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
      0,                                           /* tp_print */
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  2 warnings generated.
  g++ -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/_dispatcher.o build/temp.macosx-11.0-arm64-3.8/numba/_typeof.o build/temp.macosx-11.0-arm64-3.8/numba/_hashtable.o build/temp.macosx-11.0-arm64-3.8/numba/core/typeconv/typeconv.o -L/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/lib -lnpymath -o build/lib.macosx-11.0-arm64-3.8/numba/_dispatcher.cpython-38-darwin.so
  building 'numba._helperlib' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/cext
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/include -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/cext/utils.c
  gcc: numba/cext/dictobject.c
  gcc: numba/_helpermod.cgcc: numba/cext/listobject.c
 
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/_helpermod.o build/temp.macosx-11.0-arm64-3.8/numba/cext/utils.o build/temp.macosx-11.0-arm64-3.8/numba/cext/dictobject.o build/temp.macosx-11.0-arm64-3.8/numba/cext/listobject.o -L/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/lib -lnpymath -o build/lib.macosx-11.0-arm64-3.8/numba/_helperlib.cpython-38-darwin.so -headerpad_max_install_names
  building 'numba.core.typeconv._typeconv' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/core/typeconv/typeconv.cpp
  gcc: numba/core/typeconv/_typeconv.cpp
  g++ -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/core/typeconv/typeconv.o build/temp.macosx-11.0-arm64-3.8/numba/core/typeconv/_typeconv.o -o build/lib.macosx-11.0-arm64-3.8/numba/core/typeconv/_typeconv.cpython-38-darwin.so
  building 'numba.np.ufunc._internal' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/np
  creating build/temp.macosx-11.0-arm64-3.8/numba/np/ufunc
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/include -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/np/ufunc/_internal.c
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/np/ufunc/_internal.o -L/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/lib -lnpymath -o build/lib.macosx-11.0-arm64-3.8/numba/np/ufunc/_internal.cpython-38-darwin.so
  building 'numba.np.ufunc._num_threads' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/np/ufunc/_num_threads.c
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/np/ufunc/_num_threads.o -o build/lib.macosx-11.0-arm64-3.8/numba/np/ufunc/_num_threads.cpython-38-darwin.so
  building 'numba.mviewbuf' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/mviewbuf.c
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/mviewbuf.o -o build/lib.macosx-11.0-arm64-3.8/numba/mviewbuf.cpython-38-darwin.so -headerpad_max_install_names
  building 'numba.core.runtime._nrt_python' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/core/runtime
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/include -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/core/runtime/_nrt_pythonmod.cgcc: numba/core/runtime/nrt.c
 
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/core/runtime/_nrt_pythonmod.o build/temp.macosx-11.0-arm64-3.8/numba/core/runtime/nrt.o -L/Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/numpy/core/lib -lnpymath -o build/lib.macosx-11.0-arm64-3.8/numba/core/runtime/_nrt_python.cpython-38-darwin.so
  building 'numba.experimental.jitclass._box' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/experimental
  creating build/temp.macosx-11.0-arm64-3.8/numba/experimental/jitclass
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/experimental/jitclass/_box.c
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/experimental/jitclass/_box.o -o build/lib.macosx-11.0-arm64-3.8/numba/experimental/jitclass/_box.cpython-38-darwin.so
  building 'numba.cuda.cudadrv._extras' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  creating build/temp.macosx-11.0-arm64-3.8/numba/cuda
  creating build/temp.macosx-11.0-arm64-3.8/numba/cuda/cudadrv
  compile options: '-Inumba -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  gcc: numba/cuda/cudadrv/_extras.c
  gcc -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/cuda/cudadrv/_extras.o -o build/lib.macosx-11.0-arm64-3.8/numba/cuda/cudadrv/_extras.cpython-38-darwin.so
  building 'numba._devicearray' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  compile options: '-Inumba -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  extra options: '-std=c++11'
  gcc: numba/_devicearray.cpp
  numba/_devicearray.cpp:90:5: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
      0,                                           /* tp_print */
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  numba/_devicearray.cpp:20:5: warning: private field 'ob_base' is not used [-Wunused-private-field]
      PyObject_HEAD
      ^
  /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/object.h:81:50: note: expanded from macro 'PyObject_HEAD'
  #define PyObject_HEAD                   PyObject ob_base;
                                                   ^
  2 warnings generated.
  g++ -bundle -undefined dynamic_lookup -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -L/Users/bunnyjw/miniforge3/envs/sample/lib -arch arm64 -arch arm64 build/temp.macosx-11.0-arm64-3.8/numba/_devicearray.o -o build/lib.macosx-11.0-arm64-3.8/numba/_devicearray.cpython-38-darwin.so
  building 'numba.np.ufunc.omppool' extension
  C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64
 
  compile options: '-I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c'
  extra options: '-fopenmp -std=c++11'
  gcc: numba/np/ufunc/omppool.cpp
  gcc: numba/np/ufunc/gufunc_scheduler.cpp
  clang: error: unsupported option '-fopenmp'
  clang: error: unsupported option '-fopenmp'
  error: Command "gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include -arch arm64 -I/Users/bunnyjw/miniforge3/envs/sample/include/python3.8 -c numba/np/ufunc/gufunc_scheduler.cpp -o build/temp.macosx-11.0-arm64-3.8/numba/np/ufunc/gufunc_scheduler.o -fopenmp -std=c++11" failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for numba
  Running setup.py clean for numba
  Building wheel for llvmlite (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/bunnyjw/miniforge3/envs/sample/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"'; __file__='"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-wheel-o9giwdjt
       cwd: /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/
  Complete output (11 lines):
  running bdist_wheel
  /Users/bunnyjw/miniforge3/envs/sample/bin/python /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py
  LLVM version... Traceback (most recent call last):
    File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 220, in <module>
      main()
    File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 214, in main
      main_posix('osx', '.dylib')
    File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 134, in main_posix
      raise RuntimeError(msg) from None
  RuntimeError: Could not find a `llvm-config` binary. There are a number of reasons this could occur, please see: https://llvmlite.readthedocs.io/en/latest/admin-guide/install.html#using-pip for help.
  error: command '/Users/bunnyjw/miniforge3/envs/sample/bin/python' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for llvmlite
  Running setup.py clean for llvmlite
  Failed to build numba llvmlite
  Installing collected packages: llvmlite, numba, tslearn
    Running setup.py install for llvmlite ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/bunnyjw/miniforge3/envs/sample/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"'; __file__='"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-record-mgbrejd6/install-record.txt --single-version-externally-managed --compile --install-headers /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/llvmlite
         cwd: /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/
    Complete output (16 lines):
    running install
    /Users/bunnyjw/miniforge3/envs/sample/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running build
    got version from file /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/llvmlite/_version.py {'version': '0.37.0', 'full': 'd77dc1bcdb5af040c549f4d1ceeb4db7c8d08718'}
    running build_ext
    /Users/bunnyjw/miniforge3/envs/sample/bin/python /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py
    LLVM version... Traceback (most recent call last):
      File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 220, in <module>
        main()
      File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 214, in main
        main_posix('osx', '.dylib')
      File "/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/ffi/build.py", line 134, in main_posix
        raise RuntimeError(msg) from None
    RuntimeError: Could not find a `llvm-config` binary. There are a number of reasons this could occur, please see: https://llvmlite.readthedocs.io/en/latest/admin-guide/install.html#using-pip for help.
    error: command '/Users/bunnyjw/miniforge3/envs/sample/bin/python' failed with exit status 1
    ----------------------------------------
   ERROR: Command errored out with exit status 1: /Users/bunnyjw/miniforge3/envs/sample/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"'; __file__='"'"'/private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-install-pw3pyrl3/llvmlite_3e4d8f96420b4b3f93d9c04468170a91/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/4b/0lscybsn63l25xcyfcbjhm8c0000gn/T/pip-record-mgbrejd6/install-record.txt --single-version-externally-managed --compile --install-headers /Users/bunnyjw/miniforge3/envs/sample/include/python3.8/llvmlite Check the logs for full command output.`

**Comment by GillesVandewiele** [comment]
You might have to upgrade your pip or update your llvm:
https://github.com/numba/llvmlite/issues/621
https://github.com/numba/llvmlite/issues/531

**Comment by Mishuni** [comment]
> You might have to upgrade your pip or update your llvm: [numba/llvmlite#621](https://github.com/numba/llvmlite/issues/621) [numba/llvmlite#531](https://github.com/numba/llvmlite/issues/531)

As you mentioned, I upgrade pip using command below.
`$ python -m pip install --upgrade pip`

so, pip version is 21.3.1.
but, installing tslearn using pip still did't work.

so I installed llvmlite, llvmdev, numba using conda.

`$ conda install llvmlite`
`$ conda install -c numba llvmdev`
`$ conda install numba`

finally I got tslearn using pip.
`$ python -m pip install tslearn`

So, I can solve this problem.
Thanks
